src/pages/non-pass-finalisation/cat-cpc/__tests__/non-pass-finalisation.cat-cpc.page.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { configureTestSuite } from 'ng-bullet';
+
+describe('NonPassFinalisationCatCPCPage', () => {
+  let fixture: ComponentFixture<NonPassFinalisationCatCPCPage>;
+  let component: NonPassFinalisationCatCPCPage;
+  let store$: Store<StoreModel>;
+
+  configureTestSuite(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        NonPassFinalisationCatCPCPage,
+        MockComponent(ActivityCodeComponent),
+        MockComponent(LanguagePreferencesComponent),
+        MockComponent(DebriefWitnessedComponent),
+        MockComponent(FinalisationHeaderComponent),
+      ],
+      imports: [
+        IonicModule,
+        AppModule,
+      ],
+      providers: [
+        { provide: NavController, useFactory: () => NavControllerMock.instance() },
+        { provide: Platform, useFactory: () => PlatformMock.instance() },
+        { provide: AuthenticationProvider, useClass: AuthenticationProviderMock },
+      ],
+    });
+  });
+
+  beforeEach(async(() => {
+    fixture = TestBed.createComponent(NonPassFinalisationCatCPCPage);
+    component = fixture.componentInstance;
+    store$ = TestBed.get(Store);
+    spyOn(store$, 'dispatch');
+  }));
+
+  describe('Class', () => {
+    describe('ionViewDidEnter', () => {
+      it('should dispatch a view did enter action', () => {
+        component.ionViewDidEnter();
+        expect(store$.dispatch).toHaveBeenCalledWith(new NonPassFinalisationViewDidEnter());
+      });
+    });
+    describe('debriefWitnessedChanged', () => {
+      it('should dispatch the correct action if the inputted value is true', () => {
+        component.debriefWitnessedChanged(true);
+        expect(store$.dispatch).toHaveBeenCalledWith(new DebriefWitnessed());
+        expect(store$.dispatch).toHaveBeenCalledTimes(1);
+      });
+      it('should dispatch the correct action if the inputted value is false', () => {
+        component.debriefWitnessedChanged(false);
+        expect(store$.dispatch).toHaveBeenCalledWith(new DebriefUnwitnessed());
+        expect(store$.dispatch).toHaveBeenCalledTimes(1);
+      });
+    });
+    describe('isWelshChanged', () => {
+      it('should dispatch the correct action if the isWelsh flag is true', () => {
+        component.isWelshChanged(true);
+        expect(store$.dispatch).toHaveBeenCalledWith(new CandidateChoseToProceedWithTestInWelsh('Cymraeg'));
+        expect(store$.dispatch).toHaveBeenCalledTimes(1);
+      });
+      it('should dispatch the correct action if the isWelsh flag is false', () => {
+        component.isWelshChanged(false);
+        expect(store$.dispatch).toHaveBeenCalledWith(new CandidateChoseToProceedWithTestInEnglish('English'));
+        expect(store$.dispatch).toHaveBeenCalledTimes(1);
+      });
+    });
+    describe('OnContinue', () => {
+      it('should dispatch a change test state to WriteUp action', () => {
+        // Arrange
```

Comment:
can you remove these comments they seem a little unnecessary as it's no different to any other test

Reply:
done that